include/wil/safearrays.h

```diff
@@ -0,0 +1,897 @@
+//    PARTICULAR PURPOSE AND NONINFRINGEMENT.
+//
+//*********************************************************
+#ifndef __WIL_SAFEARRAYS_INCLUDED
+#define __WIL_SAFEARRAYS_INCLUDED
+
+#ifdef _KERNEL_MODE
+#error This header is not supported in kernel-mode.
+#endif
+
+#include <new.h> // new(std::nothrow)
+#include "wistd_type_traits.h"
+#include "resource.h" // unique_hkey
+
+namespace wil
+{
+#if defined( _OLEAUTO_H_ ) && !defined(__WIL_SAFEARRAY_)
+#define __WIL_SAFEARRAY_
+    /// @cond
+    namespace details
+    {
+        template<typename T>
+        struct var_traits{};
+
+        template<> struct var_traits<char>                       { static constexpr auto vartype = VT_I1; };
+        //template<> struct var_traits<short>                      { static constexpr auto vartype = VT_I2; };
+        template<> struct var_traits<long>                       { static constexpr auto vartype = VT_I4; };
+        template<> struct var_traits<int>                        { static constexpr auto vartype = VT_I4; };
+        template<> struct var_traits<long long>                  { static constexpr auto vartype = VT_I8; };
+        template<> struct var_traits<unsigned char>              { static constexpr auto vartype = VT_UI1; };
+        template<> struct var_traits<unsigned short>             { static constexpr auto vartype = VT_UI2; };
+        template<> struct var_traits<unsigned long>              { static constexpr auto vartype = VT_UI4; };
+        template<> struct var_traits<unsigned int>               { static constexpr auto vartype = VT_UI4; };
+        template<> struct var_traits<unsigned long long>         { static constexpr auto vartype = VT_UI8; };
+        template<> struct var_traits<float>                      { static constexpr auto vartype = VT_R4; };
+        //template<> struct var_traits<double>                     { static constexpr auto vartype = VT_R8; };
+        template<> struct var_traits<VARIANT_BOOL>               { static constexpr auto vartype = VT_BOOL; };
+        template<> struct var_traits<DATE>                       { static constexpr auto vartype = VT_DATE; };
+        template<> struct var_traits<CURRENCY>                   { static constexpr auto vartype = VT_CY; };
+        template<> struct var_traits<DECIMAL>                    { static constexpr auto vartype = VT_DECIMAL; };
+        template<> struct var_traits<BSTR>                       { static constexpr auto vartype = VT_BSTR; };
+        template<> struct var_traits<LPUNKNOWN>                  { static constexpr auto vartype = VT_UNKNOWN; };
+        template<> struct var_traits<LPDISPATCH>                 { static constexpr auto vartype = VT_DISPATCH; };
+        template<> struct var_traits<VARIANT>                    { static constexpr auto vartype = VT_VARIANT; };
+
+        inline void __stdcall SafeArrayDestroy(SAFEARRAY* psa) WI_NOEXCEPT
+        {
+            __FAIL_FAST_ASSERT__(psa != nullptr);

```

Comment:
This seems unnecessary. Just let the call do a null deref

Reply:
I'll remove the `nullptr` check from all of the API-wrappers.